cli/smartcontract/generate.go

```diff
@@ -0,0 +1,350 @@
+	if len(manifestFile) == 0 {
+		return cli.NewExitError(errNoManifestFile, 1)
+	}
+
+	manifestBytes, err := ioutil.ReadFile(manifestFile)
+	if err != nil {
+		return cli.NewExitError(fmt.Errorf("failed to read manifest file: %w", err), 1)
+	}
+
+	m := &manifest.Manifest{}
+	err = json.Unmarshal(manifestBytes, m)
+	if err != nil {
+		return cli.NewExitError(fmt.Errorf("failed to restore manifest file: %w", err), 1)
+	}
+
+	s, err := Generate(ctx, m)
+	if err != nil {
+		return cli.NewExitError(fmt.Errorf("error during generation: %w", err), 1)
+	}
+
+	err = ioutil.WriteFile(ctx.String("out"), []byte(s), os.ModePerm)
+	if err != nil {
+		return cli.NewExitError(fmt.Errorf("error during write: %w", err), 1)
+	}
+	return nil
+}
+
+func parseCallFlags(ctx *cli.Context) (map[string]string, error) {
+	flags := ctx.StringSlice("callflag")
+	result := make(map[string]string, len(flags))
+loop:
+	for i := range flags {
+		ss := strings.SplitN(flags[i], "=", 2)
+		if len(ss) == 1 {
+			return nil, fmt.Errorf("invalid callflag spec: %s", flags[i])
+		}
+		for i := range callFlags {
+			if ss[1] == callFlags[i] {
+				result[ss[0]] = ss[1]
+				continue loop
+			}
+		}
+		return nil, fmt.Errorf("invalid callflag spec: %s", flags[i])
+	}
+	return result, nil
+}
+
+type override struct {
+	Package  string
+	TypeName string
+}
+
+func parseOverrides(ctx *cli.Context) (map[string]override, error) {
+	overrides := ctx.StringSlice("override")
+	result := make(map[string]override, len(overrides))
+	for i := range overrides {
+		ss := strings.SplitN(overrides[i], "=", 2)
+		if len(ss) != 2 {
+			return nil, fmt.Errorf("invalid override spec: %s", overrides[i])
+		}
+
+		var over override
+		index := strings.LastIndexByte(ss[1], '.')
+		if index == -1 {
+			over.TypeName = ss[1]
+		} else {
+			over.Package = ss[1][:index]
+			alias := strings.LastIndexByte(ss[1], '/')
+			over.TypeName = ss[1][alias+1:]
+		}
+		result[ss[0]] = over
+	}
+	return result, nil
+}
+
+func templateFromManifest(ctx *cli.Context, m *manifest.Manifest) (contractTmpl, error) {
+	h, err := util.Uint160DecodeStringLE(ctx.String("hash"))
+	if err != nil {
+		return contractTmpl{}, fmt.Errorf("invalid hash: %w", err)
+	}
+
+	overrides, err := parseOverrides(ctx)
+	if err != nil {
+		return contractTmpl{}, err
+	}
+
+	flags, err := parseCallFlags(ctx)
+	if err != nil {
+		return contractTmpl{}, err
+	}
+
+	hStr := ""
+	for _, b := range h.BytesBE() {
+		hStr += fmt.Sprintf("\\x%02x", b)
+	}
+
+	ctr := contractTmpl{
+		PackageName:  ctx.String("package"),
+		ContractName: m.Name,
+		Hash:         hStr,
+	}
+	if ctr.PackageName == "" {
+		ctr.PackageName = lowerFirst(m.Name)
+	}
+
+	imports := make(map[string]struct{})
+	for _, m := range m.ABI.Methods {
+		if m.Name[0] == '_' {
+			continue
+		}
+
+		imports["github.com/nspcc-dev/neo-go/pkg/interop/contract"] = struct{}{}
+		imports["github.com/nspcc-dev/neo-go/pkg/interop/neogointernal"] = struct{}{}
+
+		mtd := methodTmpl{
```

Comment:
Manifest may have overloaded methods, and we don't handle this case currently. Generated code will be invalid, so I think we need an explicit check that there are no overloads and throw a warning otherwise.

Reply:
Agree, this is another argument for using main config file somehow.

Reply:
Added overload processing + tests.